hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common/src/main/java/org/apache/hadoop/yarn/server/federation/store/impl/MemoryFederationStateStore.java

```diff
@@ -464,13 +461,13 @@ public RouterMasterKeyResponse storeNewMasterKey(RouterMasterKeyRequest request)
     RouterMasterKey masterKey = request.getRouterMasterKey();
     DelegationKey delegationKey = getDelegationKeyByMasterKey(masterKey);
 
-    Set<DelegationKey> rmDTMasterKeyState = routerRMSecretManagerState.getMasterKeyState();
-    if (rmDTMasterKeyState.contains(delegationKey)) {
+    Map<Integer, DelegationKey> rmDTMasterKeyState = routerRMSecretManagerState.getMasterKeyState();
```

Comment:
I have replaced set with a map as we need to lookup objects from their unique key ids, rather than the object itself